packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/project/data/versioning/controllers/CommitRestController.java

```diff
@@ -0,0 +1,106 @@
+import org.eclipse.sirius.web.application.project.data.versioning.dto.CreateCommitRestSuccessPayload;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.GetCommitByIdRestInput;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.GetCommitByIdRestSuccessPayload;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.GetCommitChangeByIdRestInput;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.GetCommitChangeByIdRestSuccessPayload;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.GetCommitChangeRestInput;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.GetCommitChangeRestSuccessPayload;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.GetCommitsRestInput;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.GetCommitsRestSuccessPayload;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.RestCommit;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.RestDataVersion;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * REST Controller for the Project Data Versioning - Commits Endpoints.
+ *
+ * @author arichard
+ */
+@RestController
+@RequestMapping("/api/rest/projects/{projectId}/commits")
+public class CommitRestController {
+
+    private static final int TIMEOUT = 20;
+
+    private final IEditingContextDispatcher editingContextDispatcher;
+
+    public CommitRestController(IEditingContextDispatcher editingContextDispatcher) {
+        this.editingContextDispatcher = Objects.requireNonNull(editingContextDispatcher);
+    }
+
+    @GetMapping
+    public ResponseEntity<List<RestCommit>> getCommits(@PathVariable UUID projectId) {
+        var payload = this.editingContextDispatcher.dispatchQuery(projectId.toString(), new GetCommitsRestInput(UUID.randomUUID())).block(Duration.ofSeconds(TIMEOUT));
+        if (payload instanceof GetCommitsRestSuccessPayload successPayload) {
+            return new ResponseEntity<>(successPayload.commits(), HttpStatus.OK);
+        }
+        return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+    }
+
+    @PostMapping
+    public ResponseEntity<RestCommit> createCommit(@PathVariable UUID projectId, @RequestParam Optional<UUID> branchId) {
+        var payload = this.editingContextDispatcher.dispatchQuery(projectId.toString(), new CreateCommitRestInput(UUID.randomUUID(), branchId)).block(Duration.ofSeconds(TIMEOUT));
```

Comment:
dispatching a QUERY to CREATE something is odd (even if it does nothing for now)

Reply:
Done